sysl2/sysl/validate.go

```diff
@@ -0,0 +1,355 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	ERROR = 100
+	WARN  = 200
+	INFO  = 300
+	UNDEF = 400
+)
+
+type ValidationMsg struct {
+	Message string `json:"message"`
+	MsgType int    `json:"msg_type"`
+}
+
+func logMsg(messages ...ValidationMsg) {
+	for _, msg := range messages {
+		formattedMsg := "[Validator]: " + msg.Message
+		switch msg.MsgType {
+		case ERROR:
+			logrus.Error(formattedMsg)
+		case WARN:
+			logrus.Warn(formattedMsg)
+		case INFO:
+			logrus.Info(formattedMsg)
+		}
+	}
+}
+
+func noType() *sysl.Type {
+	return &sysl.Type{
+		Type: &sysl.Type_NoType_{
+			NoType: &sysl.Type_NoType{},
+		},
+	}
+}
+
+func viewOutput(syslType *sysl.Type) (string, bool) {
```

Comment:
Can you please clarify what is `viewOutput` doing?

Reply:
the method is not taking anything related to view. its argument is sysl.Type. you passing it view's return type should not decide this method's name.

Reply:
renamed to getTypeName